plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergUtil.java

```diff
@@ -595,6 +606,11 @@ public static Transaction newCreateTableTransaction(TrinoCatalog catalog, Connec
         String targetPath = getTableLocation(tableMetadata.getProperties())
                 .orElseGet(() -> catalog.defaultTableLocation(session, schemaTableName));
 
+        return catalog.newCreateTableTransaction(session, schemaTableName, schema, partitionSpec, targetPath, assemblePropertiesForCreateTable(tableMetadata));
+    }
+
+    private static Map<String, String> assemblePropertiesForCreateTable(ConnectorTableMetadata tableMetadata)
```

Comment:
Give it a more declarative name.
Maybe just "properties for create table" or "table properties for new table"?